server/src/main/java/io/deephaven/server/barrage/BarrageMessageProducer.java

```diff
@@ -110,13 +115,18 @@
      * @param <MessageView> The sub-view type that the listener expects to receive.
      */
     public interface StreamGenerator<MessageView> extends SafeCloseable {
+        interface WriteMetricsConsumer {
+            void onWrite(long bits, long cpuTime);
+        }
+
         interface Factory<MessageView> {
             /**
              * Create a StreamGenerator that now owns the BarrageMessage.
              *
              * @param message the message that contains the update that we would like to propagate
+             * @param writeTmRecorder a method that can be used to record write time
```

Comment:
Maybe update this to change the name and revise the comment (since we record data volume, too).